app/api/v2/managers/operation_api_manager.py

```diff
@@ -20,3 +48,60 @@ async def get_operation_object(self, operation_id: str, access: dict):
         if operation.match(access):
             return operation
         raise JsonHttpForbidden(f'Cannot view operation due to insufficient permissions: {operation_id}')
+
+    async def setup_operation(self, data: dict, access: BaseWorld.Access):
+        """Applies default settings to an operation if data is missing."""
+        planner_name = data.pop('planner', {}).get('name', '')
+        planner = await self._construct_planner(planner_name)
+        adversary_data = data.pop('adversary', {})
+        adversary_id = adversary_data.get('adversary_id', '')
+        adversary = await self._construct_adversary(adversary_id)
+        group_data = data.pop('host_group', '')
+        agents = await self._construct_agents(group_data)
+        sources = await self.services['data_svc'].locate('sources', match=dict(name='basic'))
+        allowed = self._get_allowed_from_access(access)
+        operation = Operation(name=data.pop('name'), id=data.pop('id', ''), planner=planner, agents=agents,
```

Comment:
Rather than relying on the constructor, we can use the marshmallow schema to our advantage by getting our dumps for the selected adversary, planner, fact source, etc, and then passing that into the marshmallow loader for operations.